src/adapter/src/coord.rs

```diff
@@ -845,6 +846,16 @@ impl PlanValidity {
                 )));
             }
         }
+
+        if let Some(create_sql) = &self.create_sql {
+            let wall_time = to_datetime((catalog.config().now)());
+            let pcx = PlanContext::new(wall_time);
+            let conn_catalog = catalog.state().for_session(session);
+            if let Err(e) = CatalogState::parse_plan(create_sql, Some(&pcx), &conn_catalog) {
```

Comment:
Although this stops the original panic from happening, it isn't sufficient for full correctness. Let's say a user was using blue-green with two schemas and each schema had a full set of their production objects. During some optimization stage the schemas are swapped. This function would not report an error because the named objects do exist, but the resulting plan would now point to other IDs. I think we need to also store the `Plan` in the `PlanValidity` and assert that both plans are identical.

Reply:
Even that is not sufficient actually. For certain statements, like `CREATE SOURCE`, we run an async stage before planning the statement. So we also need to sometimes store the `Statement<Aug>` and `ResolvedIds` and compare those to the result of re-parsing the create SQL.